python-package/lightgbm/plotting.py

```diff
@@ -433,23 +434,42 @@ def _to_graphviz(
     else:
         raise ImportError('You must install graphviz and restart your session to plot tree.')
 
-    def add(root, total_count, parent=None, decision=None):
+    def add(root, total_count, parent=None, decision=None, highlight=False):
         """Recursively add node or edge."""
+        fillcolor = 'white'
+        style = ''
+        if highlight:
+            color = 'blue'
+            penwidth = '3'
+        else:
+            color = 'black'
+            penwidth = '1'
         if 'split_index' in root:  # non-leaf
+            shape = "rectangle"
             l_dec = 'yes'
             r_dec = 'no'
             if root['decision_type'] == '<=':
-                lte_symbol = "&#8804;"
-                operator = lte_symbol
+                operator = "&#8804;"
             elif root['decision_type'] == '==':
                 operator = "="
             else:
                 raise ValueError('Invalid decision type in tree model.')
             name = f"split{root['split_index']}"
+            split_feature = root['split_feature']
             if feature_names is not None:
-                label = f"<B>{feature_names[root['split_feature']]}</B> {operator}"
+                label = f"<B>{feature_names[split_feature]}</B> {operator}"
             else:
-                label = f"feature <B>{root['split_feature']}</B> {operator} "
+                label = f"feature <B>{split_feature}</B> {operator} "
+            direction = None
+            if example_case is not None:
+                if root['decision_type'] == '==':
+                    thresholds = {int(x) for x in root['threshold'].split('||')}
+                    if example_case[split_feature] in thresholds:
+                        direction = 'left'
+                    else:
+                        direction = 'right'
+                else:
+                    direction = 'left' if example_case[split_feature] <= root['threshold'] else 'right'
```

Comment:
Does this code cover all combinations of [`use_missing`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#use_missing), [`zero_as_missing`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#zero_as_missing) params and NaN values in `example_case`?

Refer to #2921.

Reply:
I added the decisions for numerical splits in 4c19af5 and for categorical in 9f65d63